### ISSUE TITLE
Official Falco Image 0.32 contains several CRITICAL and HIGH rated CVEs (vulnerabilities)

### ISSUE THREAD

**Issue by tgip-work** [falcosecurity/falco#2108]
Hi,

I just scanned the most current docker image version `falcosecurity/falco:0.32.0` with trivy which reports several critical and high rated vulnerabilities.

**Trivy**

*Note*: This report only show the critical rated CVEs!

```text
$ trivy i -s "CRITICAL" falcosecurity/falco:0.32.0
2022-07-01T14:25:15.533+0200    INFO    Detected OS: debian
2022-07-01T14:25:15.533+0200    INFO    Detecting Debian vulnerabilities...
2022-07-01T14:25:15.563+0200    INFO    Number of language-specific files: 0

falcosecurity/falco:0.32.0 (debian 10.12)
=========================================
Total: 24 (CRITICAL: 24)

+--------------+------------------+----------+-------------------+------------------+---------------------------------------+
|   LIBRARY    | VULNERABILITY ID | SEVERITY | INSTALLED VERSION |  FIXED VERSION   |                 TITLE                 |
+--------------+------------------+----------+-------------------+------------------+---------------------------------------+
| libc-bin     | CVE-2021-33574   | CRITICAL | 2.28-10+deb10u1   |                  | glibc: mq_notify does                 |
|              |                  |          |                   |                  | not handle separately                 |
|              |                  |          |                   |                  | allocated thread attributes           |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-33574 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2021-35942   |          |                   |                  | glibc: Arbitrary read in wordexp()    |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-35942 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23218   |          |                   |                  | glibc: Stack-based buffer overflow    |
|              |                  |          |                   |                  | in svcunix_create via long pathnames  |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23218 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23219   |          |                   |                  | glibc: Stack-based buffer             |
|              |                  |          |                   |                  | overflow in sunrpc clnt_create        |
|              |                  |          |                   |                  | via a long pathname                   |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23219 |
+--------------+------------------+          +                   +------------------+---------------------------------------+
| libc-dev-bin | CVE-2021-33574   |          |                   |                  | glibc: mq_notify does                 |
|              |                  |          |                   |                  | not handle separately                 |
|              |                  |          |                   |                  | allocated thread attributes           |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-33574 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2021-35942   |          |                   |                  | glibc: Arbitrary read in wordexp()    |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-35942 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23218   |          |                   |                  | glibc: Stack-based buffer overflow    |
|              |                  |          |                   |                  | in svcunix_create via long pathnames  |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23218 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23219   |          |                   |                  | glibc: Stack-based buffer             |
|              |                  |          |                   |                  | overflow in sunrpc clnt_create        |
|              |                  |          |                   |                  | via a long pathname                   |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23219 |
+--------------+------------------+          +                   +------------------+---------------------------------------+
| libc6        | CVE-2021-33574   |          |                   |                  | glibc: mq_notify does                 |
|              |                  |          |                   |                  | not handle separately                 |
|              |                  |          |                   |                  | allocated thread attributes           |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-33574 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2021-35942   |          |                   |                  | glibc: Arbitrary read in wordexp()    |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-35942 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23218   |          |                   |                  | glibc: Stack-based buffer overflow    |
|              |                  |          |                   |                  | in svcunix_create via long pathnames  |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23218 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23219   |          |                   |                  | glibc: Stack-based buffer             |
|              |                  |          |                   |                  | overflow in sunrpc clnt_create        |
|              |                  |          |                   |                  | via a long pathname                   |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23219 |
+--------------+------------------+          +                   +------------------+---------------------------------------+
| libc6-dev    | CVE-2021-33574   |          |                   |                  | glibc: mq_notify does                 |
|              |                  |          |                   |                  | not handle separately                 |
|              |                  |          |                   |                  | allocated thread attributes           |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-33574 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2021-35942   |          |                   |                  | glibc: Arbitrary read in wordexp()    |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-35942 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23218   |          |                   |                  | glibc: Stack-based buffer overflow    |
|              |                  |          |                   |                  | in svcunix_create via long pathnames  |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23218 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23219   |          |                   |                  | glibc: Stack-based buffer             |
|              |                  |          |                   |                  | overflow in sunrpc clnt_create        |
|              |                  |          |                   |                  | via a long pathname                   |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23219 |
+--------------+------------------+          +                   +------------------+---------------------------------------+
| libc6-i386   | CVE-2021-33574   |          |                   |                  | glibc: mq_notify does                 |
|              |                  |          |                   |                  | not handle separately                 |
|              |                  |          |                   |                  | allocated thread attributes           |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-33574 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2021-35942   |          |                   |                  | glibc: Arbitrary read in wordexp()    |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2021-35942 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23218   |          |                   |                  | glibc: Stack-based buffer overflow    |
|              |                  |          |                   |                  | in svcunix_create via long pathnames  |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23218 |
+              +------------------+          +                   +------------------+---------------------------------------+
|              | CVE-2022-23219   |          |                   |                  | glibc: Stack-based buffer             |
|              |                  |          |                   |                  | overflow in sunrpc clnt_create        |
|              |                  |          |                   |                  | via a long pathname                   |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-23219 |
+--------------+------------------+          +-------------------+------------------+---------------------------------------+
| libdb5.3     | CVE-2019-8457    |          | 5.3.28+dfsg1-0.5  |                  | sqlite: heap out-of-bound             |
|              |                  |          |                   |                  | read in function rtreenode()          |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2019-8457  |
+--------------+------------------+          +-------------------+------------------+---------------------------------------+
| libssl-dev   | CVE-2022-2068    |          | 1.1.1n-0+deb10u2  | 1.1.1n-0+deb10u3 | openssl: the c_rehash script          |
|              |                  |          |                   |                  | allows command injection              |
|              |                  |          |                   |                  | -->avd.aquasec.com/nvd/cve-2022-2068  |
+--------------+                  +          +                   +                  +                                       +
| libssl1.1    |                  |          |                   |                  |                                       |
|              |                  |          |                   |                  |                                       |
|              |                  |          |                   |                  |                                       |
+--------------+                  +          +                   +                  +                                       +
| openssl      |                  |          |                   |                  |                                       |
|              |                  |          |                   |                  |                                       |
|              |                  |          |                   |                  |                                       |
+--------------+------------------+----------+-------------------+------------------+---------------------------------------+
```

**Comment by VF-mbrauer** [comment]
Any news here as this is quite important that we have with the latest release the most recent OS base image applied.

**Comment by jasondellaluce** [comment]
Hi @VF-mbrauer! The `falcosecurity/falco-no-driver` image contains only the falco binary, whereas `falcosecurity/falco` one bundles multiple old GCC versions too. This is required to support the dynamic kernel module compilation for the driver loader. In our helm charts, we separate the driver loading step and the falco executable launch step by leveraging the `falcosecurity/falco-driver-loader` and `falcosecurity/falco-no-driver` separate images (you can see `falcosecurity/falco` as a union of them). Given so, `falcosecurity/falco-no-driver` should be a more reliable source of truth for vulnerabilities concerning the Falco application.

This is a known fact in the community, but I agree this should be properly documented better.

**Comment by jasondellaluce** [comment]
Pointers to our documentation of how to run the `falco-no-driver` image:
- https://falco.org/docs/getting-started/running/#docker-least-privileged
- https://falco.org/docs/getting-started/running/#docker-privileged (the part related to`falco-no-driver` and `falco-driver-loader`)

Since version 2.0.0, our [Helm Charts](https://github.com/falcosecurity/charts/releases) rely on the `falco-no-driver` image by default and use the driver loader as an init container.

**Comment by adrianmarcu18** [comment]
> Hi @VF-mbrauer! The `falcosecurity/falco-no-driver` image contains only the falco binary, whereas `falcosecurity/falco` one bundles multiple old GCC versions too. This is required to support the dynamic kernel module compilation for the driver loader. In our helm charts, we separate the driver loading step and the falco executable launch step by leveraging the `falcosecurity/falco-driver-loader` and `falcosecurity/falco-no-driver` separate images (you can see `falcosecurity/falco` as a union of them). Given so, `falcosecurity/falco-no-driver` should be a more reliable source of truth for vulnerabilities concerning the Falco application.
> 
> This is a known fact in the community, but I agree this should be properly documented better.
What is the reason `falcosecurity/falco-driver-loader` is kept outdated? I think that even if it is used as an init container it should still be patched. There can be more to it than just polluting reports.

**Comment by deygaurab** [comment]
Opening this issue as there are medium and high CVEs on the image `falcosecurity/falco-driver-loader`

```
Vulnerabilities
===============
  CVE               Package     Version             Severity    Status                         CVSS
  ---               -------     -------             --------    ------                         ----
  CVE-2023-23916    curl        7.64.0-4+deb10u4    high        fixed in 7.64.0-4+deb10u5      7.5
  CVE-2023-0361     gnutls28    3.6.7-4+deb10u9     high        fixed in 3.6.7-4+deb10u10      7.5
  CVE-2018-10373    binutils    2.30-22             medium      fixed in 2.30.90.20180627-1    6.5
  CVE-2018-10535    binutils    2.30-22             medium      fixed in 2.30.90.20180627-1    5.5
  CVE-2018-10534    binutils    2.30-22             medium      fixed in 2.30.90.20180627-1    5.5
  CVE-2018-10372    binutils    2.30-22             medium      fixed in 2.30.90.20180627-1    5.5

```
/remove-lifecycle stale
